boto3/examples/s3.rst

```diff
@@ -0,0 +1,15 @@
+**To list objects in an Amazon S3 bucket**
+
+The following example shows how to use an Amazon S3 bucket resource to list
```

Comment:
This technically is to get attributes that are not located on the ``ObjectSummary`` but it is located on the ``Object`` resource as ``ObjectSummary`` is instantiated from ListObjects while Object is instantiated from HeadObject. I say we simplify the example or reword the sentence to be specific to the glacier example that this encapsulates or simplify the example to not have the switching from ObjectSummary to Object in order to list objects in a bucket.